sources/api/pluto/src/main.rs

```diff
@@ -141,7 +129,15 @@ use error::PlutoError;
 
 type Result<T> = std::result::Result<T, PlutoError>;
 
-async fn get_max_pods(client: &mut ImdsClient) -> Result<String> {
+async fn generate_max_pods(client: &mut ImdsClient, aws_k8s_info: &mut AwsK8sInfo) -> Result<()> {
+    if aws_k8s_info.max_pods.is_some() {
+        return Ok(());
+    }
+    aws_k8s_info.max_pods = get_max_pods(client).await.ok();
+    Ok(())
+}
+
+async fn get_max_pods(client: &mut ImdsClient) -> Result<u32> {
```

Comment:
Why split max pods into "generate" and "get" but not the others? Might be cleaner to do the same for all.

Reply:
This was to maintain the approach of not setting it should get_max_pods fails which is unique to this.